File: src/Ecommerce.Shared/Models/PedidoModel.cs
using Ecommerce.Shared.Enums;
using System.Text.Json.Serialization;

namespace Ecommerce.Shared.Models;

public sealed class PedidoModel
{
    [JsonPropertyName("identificador")]
    public Guid Identificador { get; set; }

    [JsonPropertyName("dataVenda")]
    public DateTime DataVenda { get; set; }

    [JsonPropertyName("cliente")]
    public ClienteModel Cliente { get; set; } = default!;

    [JsonPropertyName("itens")]
    public IReadOnlyCollection<PedidoItemModel> Itens { get; set; } = [];

    [JsonPropertyName("subTotal")]
    public decimal SubTotal { get; set; }

    [JsonPropertyName("desconto")]
    public decimal Desconto { get; set; }

    [JsonPropertyName("valorTotal")]
    public decimal Total { get; set; }

    [JsonPropertyName("status")]
    public PedidoStatus Status { get; set; }
}

File: src/Ecommerce.Web/src/interfaces/models/pedido.model.ts
import { PedidoStatus } from "../../enums/pedido-status.enum";
import { ClienteModel } from "./cliente.model";
import { PedidoItemModel } from "./pedido-item.model";

export interface PedidoModel {
  identificador: string;
  dataVenda: string;
  cliente: ClienteModel;
  itens: PedidoItemModel[];
  subTotal: number;
  desconto: number;
  valorTotal: number;
  status: PedidoStatus;
}
